salt/utils/win_dacl.py

```diff
@@ -1133,7 +1133,7 @@ def get_name(principal):
 
     try:
         return win32security.LookupAccountSid(None, sid_obj)[0]
-    except TypeError:
+    except (pywintypes.error, TypeError):
```

Comment:
I think `pywintypes.error` is pretty generic. Can we restrict that to the particular error we are really interested in?

Reply:
I think `pywintypes.error` is what's being returned in this case... if you look at the error in the stacktrace:
```
              error: (1332, 'LookupAccountSid', 'No mapping between account names and security IDs was done.')
```

Reply:
`pywintypes.error` is a collection of error types. Pretty sure the specific error type here is `1332`. Take a look at https://github.com/saltstack/salt/pull/44295/files for one approach to better scoping the except block.

Reply:
I would think that, since this is the only place that SID values are resolved, and no parent function is going to deal with any kind of resolution problem, this needs to be fairly generic to point to where the problem is. I don't know for sure all the ways that the mapping can go wrong, but the original problem stems from the existing exception handling being too case specific.

Reply:
Sort of. If that logic were well-accepted, you might as well just `except Exception`. You're trading one type of bad message for another type of bad message, and in the process obscuring what is different (currently unknown) exception that may have a potentially different solution.

Reply:
I wouldn't really call a more specific failure message, from the point of failure, equally bad. It would be good if there were more specific checks for the problem, but that is the end of the function and all the parent functions just seem to switch using a null SID value whenever they hit an error. If anyone knows more about the specific ways it can fail, feel free to edit.

Reply:
Can we at least include the error type in the `CommandExecutionError` that gets raised, if it exists?

Reply:
How about formatting the error if the type is pywintypes.error?
<pre>try:
    return win32security.LookupAccountSid(None, sid_obj)[0]
except (pywintypes.error, TypeError) as exc:
    if type(exc) == pywintypes.error:
        win_error = win32api.FormatMessage(exc.winerror).rstrip('\n')
        message = 'Error resolving {0} ({1})'.format(principal, win_error)
    else:
        message = 'Error resolving {0}'.format(principal)

    raise CommandExecutionError(message)</pre>
That comes out as
<pre>Error resolving S-1-5-21-419543731-3426292778-1050089031-1009 (No mapping between account names and security IDs was done)</pre>
and
<pre>Error resolving S-1-5-21-419543731-3426292778-1050089031-1009</pre>

Reply:
That seems totally reasonable.